### PR TITLE
refactor(stateBridge): The State Bridge is no longer a proxy

### PR DESCRIPTION
**Pull request by dcbuild3r**
Addresses issue #62.

### CODE REVIEW

File: src/WorldIDIdentityManagerImplV1.sol [hunk 1]
```diff
@@ -81,7 +82,7 @@ contract WorldIDIdentityManagerImplV1 is WorldIDImpl, IWorldID {
     SemaphoreVerifier internal semaphoreVerifier;
 
     /// @notice The interface of the bridge contract from L1 to supported target chains.
-    address internal _stateBridgeProxyAddress;
+    address internal _stateBridgeAddress;
```
Comment (iamrecursion) [inline src/WorldIDIdentityManagerImplV1.sol:85]: This should just be `IBridge` now. We were only taking the `address` due to it being a proxy.

File: src/WorldIDIdentityManagerImplV1.sol [hunk 2]
```diff
@@ -553,50 +551,45 @@ contract WorldIDIdentityManagerImplV1 is WorldIDImpl, IWorldID {
...
         }
     }
 
-    /// @notice Allows a caller to query the address of the current stateBridgeProxy.
+    /// @notice Allows a caller to query the address of the current StateBridge.
     ///
-    /// @return proxy The address of the currently used stateBridgeProxy
-    function stateBridgeProxyAddress()
+    /// @return proxy The address of the currently used StateBridge
+    function stateBridgeAddress()
```
Comment (iamrecursion) [inline src/WorldIDIdentityManagerImplV1.sol:562]: Should just be `stateBridge() returns (IBridge bridge)`.

File: src/WorldIDIdentityManagerImplV1.sol [hunk 3]
```diff
@@ -553,50 +551,45 @@ contract WorldIDIdentityManagerImplV1 is WorldIDImpl, IWorldID {
...
     }
 
-    /// @notice Allows a caller to upgrade the stateBridgeProxy.
+    /// @notice Allows a caller to upgrade the stateBridge.
     /// @dev Only the owner of the contract can call this function.
     ///
-    /// @param newStateBridgeProxyAddress The address of the new stateBridgeProxy
-    function setStateBridgeProxyAddress(address newStateBridgeProxyAddress)
+    /// @param newStateBridgeAddress The address of the new stateBridge
+    function setStateBridgeAddress(address newStateBridgeAddress)
```
Comment (iamrecursion) [inline src/WorldIDIdentityManagerImplV1.sol:577]: Should just be `setStateBridge(IBridge newStateBridge)`.

**Review by iamrecursion — CHANGES_REQUESTED** [review]
This needs to be done in a more type-safe manner now that we can.

**Comment by dcbuild3r** [comment]
Fixed all the issues brought up in review

File: src/WorldIDIdentityManagerImplV1.sol [hunk 4]
```diff
@@ -85,7 +86,7 @@ contract WorldIDIdentityManagerImplV1 is WorldIDImpl, IWorldID {
     ISemaphoreVerifier internal semaphoreVerifier;
 
     /// @notice The interface of the bridge contract from L1 to supported target chains.
-    address internal _stateBridgeProxyAddress;
+    IBridge _stateBridge;
```
Comment (iamrecursion) [inline src/WorldIDIdentityManagerImplV1.sol:89]: Why is this no longer `internal`?

File: src/test/identity-manager/WorldIDIdentityManagerTest.sol [hunk 5]
```diff
@@ -54,9 +55,9 @@ contract WorldIDIdentityManagerTest is WorldIDTest {
     uint256 internal constant SNARK_SCALAR_FIELD =
         21888242871839275222246405745257275088548364400416034343698204186575808495617;
 
-    // StateBridgeProxy mock
-    SimpleStateBridge internal stateBridge;
-    address internal stateBridgeProxy;
+    // StateBridgemock
+    SimpleStateBridge internal simpleStateBridge;
+    IBridge internal stateBridge;
```
Comment (iamrecursion) [inline src/test/identity-manager/WorldIDIdentityManagerTest.sol:60]: Do you ever need it to be typed as a `SimpleStateBridge`? As best I can tell we only ever use it as `IBridge`, so this two-layer construction is unnecessary.

**Review by iamrecursion — CHANGES_REQUESTED** [review]
Good stuff with the type safety, but it still needs a few changes.

File: src/test/identity-manager/WorldIDIdentityManagerTest.sol (hunk 5 above)
Comment (dcbuild3r) [reply src/test/identity-manager/WorldIDIdentityManagerTest.sol:60]: well I just need a mock which I can later call through an interface

File: src/WorldIDIdentityManagerImplV1.sol (hunk 4 above)
Comment (dcbuild3r) [reply src/WorldIDIdentityManagerImplV1.sol:89]: fixed in next commit

**Pull request merged**